app/core/Encryptor/pbkdf2.ts

```diff
@@ -0,0 +1,32 @@
+import { bytesToString, hexToBytes } from '@metamask/utils';
+import { NativeModules } from 'react-native';
+import { ShaAlgorithm } from './constants';
+
+/**
+ * Derives a key using PBKDF2.
+ *
+ * @param password - The password used to generate the key.
+ * @param salt - The salt used during key generation.
+ * @param iterations - The number of iterations used during key generation.
+ * @param keyLength - The length of the key to generate.
+ * @returns The generated key.
+ */
+const pbkdf2 = async (
+  password: Uint8Array,
+  salt: Uint8Array,
+  iterations: number,
+  keyLength: number,
+): Promise<Uint8Array> => {
+  const Aes = NativeModules.Aes;
+  const derivedKey = await Aes.pbkdf2(
+    bytesToString(password),
+    bytesToString(salt),
+    iterations,
+    keyLength * 8,
```

Comment:
Also being explicit here:

```suggestion
    keyLength * 8, // bits
```

Reply:
Probably cleaner to have a function for converting bytes to bits so it's more clear

Reply:
Added the function in this [commit](https://github.com/MetaMask/metamask-mobile/pull/12428/commits/2666dbec48ac3d0d5ecdcc97306e358702c7de99) and respective unit test coverage <3